faiss/gpu/GpuIndexIVFPQ.cu

```diff
@@ -453,5 +457,26 @@ void GpuIndexIVFPQ::verifySettings_() const {
             requiredSmemSize);
 }
 
+size_t GpuIndexIVFPQ::remove_ids(const IDSelector& sel) {
+    const IDSelectorArray* sela = dynamic_cast<const IDSelectorArray*>(&sel);
+    if (sela->n <= 0 || !sela->ids) {
```

Comment:
throw an error if `sela` is NULL (i.e., if an IDSelectorArray instance wasn't passed)